batch.go

```diff
@@ -81,6 +81,7 @@ func (batch *Batch) close() (err error) {
 	batch.lock = nil
 	if batch.msgs != nil {
 		batch.msgs.discard()
+		releaseBuffer(batch.msgs.decompressed)
```

Comment:
Could we change this to set the decompressed field to `nil` after releasing it? Otherwise there's a risk to releasing the buffer again if Close is called multiple times.

```go
if batch.msgs.decompressed != nil {
   releaseBuffer(batch.msgs.decompressed)
  batch.msgs.decompressed = nil
}
```

Reply:
Good catch. Fixed now.